### PR TITLE
move org essentials to spacemacs distro

### PR DESCRIPTION
**Pull request by d12frosted**
Following #4277 

---

I haven't touched readme file, because I think it's good to have all description in one place. Also, I didn't rename `org` layer to anything else. If you think that it's a good idea - just tell me (or it could be done via another PR / commit). 

I tested my changes in different situations and everything seems to be working. But since I don't know all use-cases of org layer, I would like to take some help from org users and those who don't use org at all. 

In case you think that something is not in the right place (some configs should be moved from spacemacs-org to org or vice versa) just let's discuss :)

### CODE REVIEW

**Comment by JAremko** [comment]
:+1:  I don't understand why org isn't a part of Spacemacs. Pretty much everyone will want to read README.org files for layers.  And if we going to fix #4399 TOC will look  rather ugly in the fundamental mode.

```
* Spacemacs documentation                                   :TOC_4_org:noexport:
 - [[#core-pillars][Core Pillars]]
   - [[#mnemonic][Mnemonic]]
   - [[#discoverable][Discoverable]]
   - [[#consistent][Consistent]]
   - [[#crowd-configured][Crowd-Configured]]
 - [[#highlighted-feature][Highlighted feature]]
 - [[#screenshots][Screenshots]]
 - [[#who-can-benefit-from-this][Who can benefit from this?]]
```

**Comment by a13ph** [comment]
to play devils advocate, orgmode is still a plaintext, readable fine even on paper or in any plain text editor. navigation somewhat suffers, and it's not as pretty, but it's not hell at all

That said, I agree that at least some support for orgmode is needed closer to the core - it's one of most popular packages and reasons to migrate, as well as currently one of the best documentation formats for emacs users

**Comment by d12frosted** [comment]
> to play devils advocate, orgmode is still a plaintext, readable fine even on paper or in any plain text editor. navigation somewhat suffers, and it's not as pretty, but it's not hell at all

I read and edit org files on iPad as plain text files :/

File: layers/+spacemacs/spacemacs-org/packages.el [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      ;;     "Try removing org code from user initialization and private layers.") t))
+
+      (setq org-startup-with-inline-images t
+            org-src-fontify-natively t
+            ;; this is consistent with the value of
+            ;; `helm-org-headings-max-depth'.
+            org-imenu-depth 8)
+
+      (with-eval-after-load 'org-indent
+        (spacemacs|hide-lighter org-indent-mode))
```
Comment (syl20bnr) [inline layers/+spacemacs/spacemacs-org/packages.el]: Now we disable org-indent by default we can remove this line.

File: layers/+spacemacs/spacemacs-org/packages.el [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+      ;; Add global evil-leader mappings. Used to access org-agenda
+      ;; functionalities – and a few others commands – from any other mode.
+      (spacemacs/declare-prefix "ao" "org")
```
Comment (syl20bnr) [inline layers/+spacemacs/spacemacs-org/packages.el]: Should be in `org` layer if there is no key bindings under this prefix in `spacemacs-org`

**Comment by syl20bnr** [comment]
Nice work, the documentation will need to be split at some point but it can wait because this is something we have to rethink given we have multiple spacemacs layers now.

File: layers/+spacemacs/spacemacs-org/packages.el (hunk 1 above)
Comment (d12frosted) [inline layers/+spacemacs/spacemacs-org/packages.el]: Ah. Forgot that we don't enable it anymore. Good point.

**Comment by d12frosted** [comment]
@syl20bnr thanks for review - fixed what you've asked.

**Comment by d12frosted** [comment]
Ping, ping :cry:

**Comment by JAremko** [comment]
This branch has conflicts that must be resolved :shipit:

**Comment by d12frosted** [comment]
Ok, updated this one as well. Would be nice to have it merged as soon as possible :)

**Comment by JAremko** [comment]
@syl20bnr  :shipit:

**Comment by TheBB** [comment]
Now that we can sort PRs by :+1: reactions on the first post, that is a handy way for people to signal which PRs are most interesting to them. I've just added mine.

**Comment by JAremko** [comment]
And two goes from me

**Comment by nixmaniack** [comment]
> Now that we can sort PRs by :+1: reactions on the first post, that is a handy way for people to signal which PRs are most interesting to them.

This should be mentioned somewhere so that everyone is aware.

**Comment by JAremko** [comment]
Also we can sort by the amount of confusion caused - neat.

**Comment by d12frosted** [comment]
> Don't forger to selfthumb

@JAremko just did it :) 

But do I really want to see it in Spacemacs? Hm. I wonder.

**Comment by syl20bnr** [comment]
Thank you ! :+1: 
I rearranged it a bit in commit https://github.com/syl20bnr/spacemacs/commit/ff7b839
Cherry-picked into develop branch, you can safely delete your branch.

**Pull request closed without merging**